### PR TITLE
Implement player board hiding

### LINKED ISSUE #52: Feature: choose player from player list to hide the game board from.
For fast money we have to pick two team mates from 1 team. 
We have to tell 1 player to "leave the room" or video call while we run the 1st fast money round. 

While they leave the room the game board is still visible on their joined device. 
Most people in spirit would choose not to look, however the game should up hold the rules of the game.

It would be nice if the admin had a choice on who to "hide" the game board from during fast money.

### PR DESCRIPTION
**Pull request by karlromets**
Resolves #52 

**Description:**
Allows for player buzzer screen hiding from host panel

**Changes made:**
- Before deleting registeredPlayers in ws.js, we save the hidden states and add them to the clone
- Buzzer shows only text of "You have been blindfolded..." if hidden = true
- Toggle button is in admin.js playerlist next to the kick button, uses lucide's Eye & EyeOff icon

**Notes:**
The player list is duplicated in both normal and final round views (final round view originally didn't have a player list), should restructure the admin panel a bit so that the player list is in one spot

### CODE REVIEW

File: components/buzzer.js [hunk 1]
```diff
@@ -117,6 +117,10 @@ export default function Buzzer(props) {
     });
   }, []);
 
+
+  const currentPlayer = game.registeredPlayers[props.id];
+  if (currentPlayer?.hidden) return <h1>{t("You have been blindfolded...")}</h1>
```
Comment (joshzcold) [inline components/buzzer.js:122]: How about this to center the text?
```jsx
  if (currentPlayer?.hidden)
    return (
      <div className="flex flex-col justify-center items-center min-h-screen">
        <h1>{t("You have been blindfolded...")}</h1>
      </div>
    );

```

**Review by joshzcold — CHANGES_REQUESTED** [review]
Just one small change and this is a ready I believe.

This is a great new feature. Thank you 👍

File: pages/api/ws.js [hunk 2]
```diff
@@ -374,8 +375,30 @@ const ioHandler = (req, res) => {
...
+                // Skip if player is "host" or not an object
+                if (typeof player === 'object' && player !== null && typeof player.hidden === 'boolean') {
+                  incomingHiddenStates[id] = { hidden: player.hidden };
+                }
+              });
+            }
+
+            // Remove registered players data from client message to prevent unauthorized player modifications
             delete message.data.registeredPlayers;
+
```
Comment (joshzcold) [inline pages/api/ws.js:392]: Needs to be implemented in the new backend.

About here is where the player struct object is so you can add a new field.
https://github.com/joshzcold/Cold-Family-Feud/blob/4ddc2088d2ceec30c11ce453fe7808a32a8ea883/backend/api/game.go#L13

Here is where you can add new actions
https://github.com/joshzcold/Cold-Family-Feud/blob/4ddc2088d2ceec30c11ce453fe7808a32a8ea883/backend/api/pipe.go#L28

Here is where a player is created.
NOTE: in golang structs the default value if not specified for boolean would be `false` automatically.
https://github.com/joshzcold/Cold-Family-Feud/blob/4ddc2088d2ceec30c11ce453fe7808a32a8ea883/backend/api/session.go#L186

Comment (joshzcold) [reply pages/api/ws.js:392]: The function to handle the generic `data` action is here
https://github.com/joshzcold/Cold-Family-Feud/blob/4ddc2088d2ceec30c11ce453fe7808a32a8ea883/backend/api/data.go#L24

**Review by joshzcold — CHANGES_REQUESTED** [review]
Needs to be put into the new backend.

File: pages/api/ws.js (hunk 2 above)
Comment (karlromets) [reply pages/api/ws.js]: Added Hidden field now works same as with old backend

**Review by joshzcold — APPROVED** [review]

**Pull request merged**